core-bundle/contao/controllers/BackendMain.php

```diff
@@ -249,6 +251,22 @@ protected function compileTemplateData(array $data): array
 		$data['menu'] = $twig->render('@ContaoCore/Backend/be_menu.html.twig');
 		$data['headerMenu'] = $twig->render('@ContaoCore/Backend/be_header_menu.html.twig');
 
+		$responseContext = System::getContainer()->get('contao.routing.response_context_accessor')->getResponseContext();
+
+		if ($responseContext?->has(HtmlHeadBag::class))
+		{
+			$htmlHeadBag = $responseContext->get(HtmlHeadBag::class);
+			$data['metaTags'] = array_combine(
+				array_map('Contao\StringUtil::specialcharsAttribute', array_keys($htmlHeadBag->getMetaTags())),
+				array_map('Contao\StringUtil::specialcharsAttribute', array_values($htmlHeadBag->getMetaTags()))
```

Comment:
You could use first class callable syntax here. :slightly_smiling_face: 
```suggestion
				array_map(Contao\StringUtil::specialcharsAttribute(...), array_keys($htmlHeadBag->getMetaTags())),
				array_map(Contao\StringUtil::specialcharsAttribute(...), array_values($htmlHeadBag->getMetaTags()))
```

But shouldn't escaping be done in the template? The HTML Attributes class also already escapes values and validates names, so in this case nothing is needed IMHO.

Reply:
In legacy templates we escape in the code most of the time. Due to input encoding it is expected that legacy template variables are always encoded, depending on the use case.